### PR TITLE
run pylint in format action

### PR DESCRIPTION
**Pull request by timmysilv**
the current pylint action uses [ricardochaves/python-lint](https://github.com/ricardochaves/python-lint) which is unmaintained. I searched around, and realized there are no actively-maintained pylint actions! This is a concern for one main reason: that action is dockerized and only runs with python 3.9. Unfortunately, `pylint` gives different results with different python versions. One day, we might not be able to lint pennylane using python 3.9 (but we probably could - it's not like `pylint` needs to run any code!), and we'd be stuck!

I started linting some files using the latest version of `pylint`, but I realized that there are way more files with issues than I thought so I just pinned it to the current version. We can start addressing those later. ~Lmk if you want me to revert the changes and just deal with it all later.~ EDIT: I have reverted the changes.

To be clear, there are 2 different things that will need upgrading (not in this PR, but later):
1. switching to the latest version of pylint
2. linting with a newer version of Python

PS: I tried to just use `pylint pennylane/**/*.py` as [suggested](https://github.com/pylint-dev/pylint/pull/2758#issuecomment-561164761) by a potential pylint contributor, but found that `**` evaluates differently in different shells (eg. old bash vs. new bash vs. zsh) and didn't do what I wanted on the `ubuntu-latest` runner

PPS: I will need someone with access to change the branch protection to accommodate however this ends up

### CODE REVIEW

File: tests/numpy/test_numpy_wrapper.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class TestExtractTensors:
     def test_empty_terable(self):
         """Test that an empty iterable returns nothing"""
         res = list(np.extract_tensors([]))
-        assert res == []
+        assert not res
```
Comment (albi3ro) [inline tests/numpy/test_numpy_wrapper.py:34]:
```suggestion
        assert res == [] # pylint: disable=use-implicit-booleaness-not-comparison
```
but I might have spelled booleaness wrong.

Basically, asserting `res` is falsy is not the same as assert that it's an empty list.  The output could have been `False` for some reason, and we would need to catch that.

**Comment by albi3ro** [comment]
Nice solution.  I'm just wondering if there's some way we can validate that the action fails when we need it to fail.

File: tests/numpy/test_numpy_wrapper.py (hunk 1 above)
Comment (timmysilv) [reply tests/numpy/test_numpy_wrapper.py:34]: generally I agree, but the line right above it says `res = list(...)`, so we know it's a list object. the only way a list object is falsy is if it's empty

**Comment by timmysilv** [comment]
what do you mean? if any of black, pylint, or pylint-test fails, the action should fail. i just wanted to ensure that all 3 of them run so you don't have to learn about pylint errors after fixing black errors

**Comment by trbromley** [comment]
> Lmk if you want me to revert the changes and just deal with it all later.

I'd say its easier to keep these changes separate from the CI updates in this PR, but will leave it up to the reviewers.

**Review by albi3ro — APPROVED** [review]

**Review by PietropaoloFrisoni — APPROVED** [review]
Nice! Thanks for this : )

**Pull request merged**